plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/UriTemplateServerModifier.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.machine;
+
+import org.eclipse.che.api.machine.server.model.impl.ServerImpl;
+import org.slf4j.Logger;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static java.lang.String.format;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Modifies machine server attributes according to provided template of URI of the server.
+ *
+ * @author Alexander Garagatyi
+ */
+public abstract class UriTemplateServerModifier implements MachineServerModifier {
+    private static final Logger LOG = getLogger(RemoteDockerNode.class);
+
+    private String serverUrlTemplate;
```

Comment:
you never change the `serverUrlTemplate` value, i think you cab define it as final, and then all the stateless classes which extend this class may be `@Singleton` annotated.